packages/datadog-instrumentations/src/fs.js

```diff
@@ -0,0 +1,357 @@
+  wrap(fs.promises.realpath, 'native', createWrapFunction('', 'realpath.native'))
+
+  wrap(fs, 'createReadStream', createWrapCreateStream())
+  wrap(fs, 'createWriteStream', createWrapCreateStream())
+  if (fs.Dir) {
+    wrap(fs.Dir.prototype, 'close', createWrapFunction('dir.'))
+    wrap(fs.Dir.prototype, 'closeSync', createWrapFunction('dir.'))
+    wrap(fs.Dir.prototype, 'read', createWrapFunction('dir.'))
+    wrap(fs.Dir.prototype, 'readSync', createWrapFunction('dir.'))
+    wrap(fs.Dir.prototype, Symbol.asyncIterator, createWrapDirAsyncIterator())
+  }
+
+  wrap(fs, 'unwatchFile', creaetWatchWrapFunction())
+  wrap(fs, 'watch', creaetWatchWrapFunction())
+  wrap(fs, 'watchFile', creaetWatchWrapFunction())
+
+  wrapFileHandle(fs)
+  return fs
+})
+
+async function wrapFileHandle (fs) {
+  const fileHandlePrototype = await getFileHandlePrototype(fs)
+  const desc = Reflect.getOwnPropertyDescriptor(fileHandlePrototype, kHandle)
+  if (!desc || !desc.get) {
+    Reflect.defineProperty(fileHandlePrototype, kHandle, {
+      get () {
+        return this[ddFhSym]
+      },
+      set (h) {
+        this[ddFhSym] = h
+        wrap(this, 'close', createWrapFunction('filehandle.'))
+      },
+      configurable: true
+    })
+  }
+  for (const name of Reflect.ownKeys(fileHandlePrototype)) {
+    if (typeof name !== 'string' || name === 'constructor' || name === 'fd' || name === 'getAsyncId') {
+      continue
+    }
+    wrap(fileHandlePrototype, name, createWrapFunction('filehandle.'))
+  }
+}
+
+async function getFileHandlePrototype (fs) {
+  const fh = await fs.promises.open(__filename, 'r')
+  if (!kHandle) {
+    kHandle = Reflect.ownKeys(fh).find(key => typeof key === 'symbol' && key.toString().includes('kHandle'))
+  }
+  fh.close()
+
+  return Object.getPrototypeOf(fh)
+}
+
+function getSymbolName (sym) {
+  return sym.description || sym.toString()
+}
+function initDirAsyncIteratorProperties (iterator) {
+  const keys = Reflect.ownKeys(iterator)
+  for (const key of keys) {
+    if (kDirReadPromisified && kDirClosePromisified) break
+    if (typeof key !== 'symbol') continue
+    if (!kDirReadPromisified && getSymbolName(key).includes('kDirReadPromisified')) {
+      kDirReadPromisified = key
+    }
+    if (!kDirClosePromisified && getSymbolName(key).includes('kDirClosePromisified')) {
+      kDirClosePromisified = key
+    }
+  }
+}
+
+function createWrapDirAsyncIterator () {
+  return function wrapDirAsyncIterator (asyncIterator) {
+    return function wrappedAsyncIterator () {
+      if (!kDirReadPromisified || !kDirClosePromisified) {
+        initDirAsyncIteratorProperties(this)
+      }
+      wrap(this, kDirReadPromisified, createWrapFunction('dir.', 'read'))
+      wrap(this, kDirClosePromisified, createWrapFunction('dir.', 'close'))
+      return asyncIterator.apply(this, arguments)
+    }
+  }
+}
+
+function createWrapCreateStream () {
+  return function wrapCreateStream (original) {
```

Comment:
Why wrap this in a closure when you're not passing anything in? You could just reference `wrapCreateStream` directly wherever you would call `createWrapCreateStream`.

Reply:
You're right. changed.